dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/IdentifierRestRepository.java

```diff
@@ -0,0 +1,308 @@
+ * previously implmented in org.dspace.app.rest.IdentifierRestController
+ *
+ * @author Kim Shepherd
+ */
+@RestController
+@RequestMapping("/api/" + IdentifierRestRepository.CATEGORY)
+@Component(IdentifierRestRepository.CATEGORY + "." + IdentifierRestRepository.NAME)
+public class IdentifierRestRepository extends DSpaceRestRepository<IdentifierRest, String> implements InitializingBean {
+    @Autowired
+    private DiscoverableEndpointsService discoverableEndpointsService;
+    @Autowired
+    private UriListHandlerService uriListHandlerService;
+    @Autowired
+    private DOIService doiService;
+    @Autowired
+    private HandleService handleService;
+    @Autowired
+    private ItemService itemService;
+
+    // Set category and name for routing
+    public static final String CATEGORY = "pid";
+    public static final String NAME = IdentifierRest.NAME;
+
+    /**
+     * Register /api/pid/find?id=... as a discoverable endpoint service
+     *
+     * @throws Exception
+     */
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        discoverableEndpointsService
+                .register(this,
+                        Arrays.asList(Link.of(UriTemplate.of("/api/pid/find",
+                                                new TemplateVariables(
+                                                        new TemplateVariable("id",
+                                                                TemplateVariable.VariableType.REQUEST_PARAM))),
+                                        CATEGORY)));
+    }
+
+    /**
+     * Find all identifiers. Not implemented.
+     * @param context
+     *            the dspace context
+     * @param pageable
+     *            object embedding the requested pagination info
+     * @return
+     */
+    @PreAuthorize("permitAll()")
+    @Override
+    public Page<IdentifierRest> findAll(Context context, Pageable pageable) {
+        throw new RepositoryMethodNotImplementedException(IdentifierRest.NAME, "findAll");
+    }
+
+    /**
+     * Find the identifier object for a given identifier string (eg. doi).
+     * Not implemented -- Tomcat interprets %2F as path separators which means
+     * parameters are a safer way to handle these operations
+     *
+     * @param context
+     *            the dspace context
+     * @param identifier
+     *            the rest object id
+     * @return
+     */
+    @PreAuthorize("permitAll()")
+    @Override
+    public IdentifierRest findOne(Context context, String identifier) {
+        throw new RepositoryMethodNotImplementedException(IdentifierRest.NAME, "findOne");
+    }
+
+    /**
+     * Find identifiers associated with a given item
+     * @param uuid
+     * @param pageable
+     * @return
+     */
+    @SearchRestMethod(name = "findByItem")
+    @PreAuthorize("permitAll()")
+    public Page<IdentifierRest> findByItem(@Parameter(value = "uuid", required = true)
```

Comment:
I think we may be missing ITs for `IdentifierRestRepository`?  I'm not seeing an `IdentifierRestRepositoryIT` class.  Probably you should rename the old `IdentifierRestControllerIT` class to be `IdentifierRestRepositoryIT` (since you replaced IdentifierRestController).  Then, I'd recommend adding new tests for:
* findByItem
* creating (minting and queuing)

Reply:
Yes, I refactored to `IdentifierRestRepositoryIT` which now tests these methods, as well as the pid/find methods used in the legacy controller